File: services/graphql-service/app/application/use_cases/user/create_user_use_case.py
from app.domain.entities.user import User
from app.application.interfaces.user_repository_interface import IUserRepository
from app.application.dto.user_response import UserResponse

class CreateUserUseCase:

    def __init__(
        self,
        repository: IUserRepository
    ):
        self.repository = repository

    async def execute(
        self,
        user: User
    ) -> UserResponse:
        
        return await self.repository.create(
            user
        )

File: services/grpc-service/src/GrpcService.Application/UseCases/CreateUser/CreateUserUseCase.cs


using GrpcService.Application.Dtos;
using GrpcService.Application.Interfaces;
using GrpcService.Application.Mappers;
using GrpcService.Domain.Entities;

namespace GrpcService.Application.UseCases.CreateUser;

public sealed class CreateUserUseCase(IUserRepository repository)
{
    public async Task<UserDto> ExecuteAsync(
        CreateUseDto dto,
        CancellationToken cancellationToken = default
    )
    {
        var user = User.Create(dto.Name, dto.Email, dto.Age);

        var created = await repository.CreateAsync(user, cancellationToken);

        return created.ToDto();
    }
}